src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -137,6 +138,59 @@ private string ResolveMatchedPath(string matchedPath)
             return RequestPathBase.Add(relativePath).ToString();
         }
 
+        private class PathComparer : IComparer<string>
+        {
+            public int Compare(string x, string y)
+            {
+                // < 0 = x < y
+                // > 0 = x > y
+
+                if (x.Equals(y))
+                {
+                    return 0;
+                }
+
+                var xExtIndex = x.LastIndexOf('.');
+                var yExtIndex = y.LastIndexOf('.');
```

Comment:
No, as this is only used to sort a result from `Matcher` which never returns null entries. Of course, if it's used somewhere else, then yes. I can add an explicit check just to be super sure.